benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <benchmark/benchmark.h>
+#include <spdlog/spdlog.h>
+#include <opencv2/opencv.hpp>
+#include <image_event_stamped.h>
+#include <klepsydra/mem_core/basic_middleware_provider.h>
+#include <klepsydra/vision_ocv/image_data_factory.h>
+
+
+
+static void BM_FactoryPerformanceTest(benchmark::State &state)
+{
+    // vars to test
+    bool withFactory = state.range(0);
+
+    kpsr::vision_ocv::ImageDataFactory imageDataFactory(500, 500, cv::IMREAD_UNCHANGED, "raw_image");
+    auto initF = [&imageDataFactory](kpsr::vision_ocv::ImageDataStamped & e) {
+        e.stamp = std::chrono::high_resolution_clock::now();
+        imageDataFactory.initializerFunction(e.im);
+    };
+    auto cloneF = [&imageDataFactory] (const kpsr::vision_ocv::ImageDataStamped &source,
+                                                     kpsr::vision_ocv::ImageDataStamped &target) {
+        target.stamp = source.stamp;
+        imageDataFactory.eventClonerFunction(source.im, target.im);
+    };
+    double total_duration = 0;
+    std::unique_ptr<kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageDataStamped>> mp;
+    if (withFactory) {
+        mp.reset(new kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageDataStamped>(nullptr,
+          "img", 4, 6, initF, cloneF, false));
+    } else {
+        mp.reset(new kpsr::mem::BasicMiddlewareProvider<kpsr::vision_ocv::ImageDataStamped>(nullptr,
+          "img", 4, 6, nullptr, nullptr, false));
+    }
+    mp->start();
+    auto receiveImGetDuration = [&total_duration](const kpsr::vision_ocv::ImageDataStamped & im) {
+        auto now = std::chrono::high_resolution_clock::now();
+        auto duration = std::chrono::duration_cast<std::chrono::duration<double>>(now - im.stamp).count();
+        total_duration += duration; // maybe should be atomic
+    };
+    auto subscriber = mp->getSubscriber();
+    subscriber->registerListener("receiveIm", receiveImGetDuration);
+
+    // Pub stuff
+    kpsr::vision_ocv::ImageDataStamped event;
+    initF(event);
+    auto publisher = mp->getPublisher();
+    // stats
+    auto &publisherStats = publisher->_publicationStats;
+    auto listenerStats = subscriber->getSubscriptionStats("receiveIm");
+
+    for (auto _ : state) {
+        publisher->publish(event);
```

Comment:
What's happening is that without the factory, when the initF is empty, the timestamp for the event creation is never updated. So before you publish, you should update `event.stamp` here.